src/payment/strategies/ppsdk/step-handler/continue-handler/redirect/RedirectionState.ts

```diff
@@ -0,0 +1,43 @@
+export const PENDING_REDIRECT_PARAM = 'redirecting';
+
+export class RedirectionState {
+    private _isRedirecting: boolean;
+
+    constructor() {
+        this._isRedirecting = this.getCurrentUrl().searchParams.has(PENDING_REDIRECT_PARAM);
+    }
+
+    private getCurrentUrl(): URL {
+        return new URL(window.location.href);
+    }
+
+    private replaceCurrentUrl(url: URL) {
+        window.history.replaceState(null, '', url.href);
+    }
+
+    private setRedirectingParamOnUrl() {
+        const url = this.getCurrentUrl();
+        url.searchParams.set(PENDING_REDIRECT_PARAM, 'true');
+        this.replaceCurrentUrl(url);
+    }
+
+    private removeRedirectingParamFromUrl() {
+        const url = this.getCurrentUrl();
+        url.searchParams.delete(PENDING_REDIRECT_PARAM);
+        this.replaceCurrentUrl(url);
+    }
+
+    get isRedirecting() {
+        return this._isRedirecting;
+    }
+
+    set isRedirecting(value: boolean) {
```

Comment:
We should probably make this a linter rule. But we follow this rule (https://github.com/airbnb/javascript#accessors--no-getters-setters) in our project. In this case, it's not clear that setting `true/false` on an object actually produces the side effect of changing the browser history. It'd be much clear if methods with intention-revealing names are called instead (e.g.: `setRedirectingParamOnUrl`, `removeRedirectingParamFromUrl`).

Reply:
Agreed on dropping the custom getter/setter - they are too 'magic': https://github.com/bigcommerce/checkout-sdk-js/pull/1285/commits/6954c7433dd2cd3f2cb15fd72a6e6d5df8eccfb1

I've stuck with `isRedirecting` and `setRedirecting` for the method names for now.

The idea was not to expose the url based implementation detail of the `RedirectionState` class to consumers. E.g. the consumer wants the `RedirectionState` to keep track of redirect activity, it is not concerned or opinionated about _how_ `RedirectionState` keeps that state.

Reply:
Sure I'm good with that 👍